filter.php

```diff
@@ -558,22 +559,20 @@ public function filter($text, array $options = array()) {
             // Open that as a new doc to pull the video node out.
             $tempdom = new DOMDocument('1.0', 'UTF-8');
             @$tempdom->loadHTML($newtext, LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD);
-            $newvideo = $tempdom->getElementsByTagName('video')[0];
+            $xpath = new DomXPath($tempdom);
+            $query = $xpath->query("//*[contains(@class, 'local-smartmedia-wrapper')]");
 
-            if (is_null($newvideo)) {
-                // We must have replaced it with a placeholder. Use that as the new "video".
-                $newvideo = $tempdom->firstChild;
+            if (count($query) === 0) {
```

Comment:
> a DOMNodeList containing all nodes matching the given XPath expression. Any expression which does not return nodes will return an empty DOMNodeList. The return is false if the expression is malformed or the contextnode is invalid.

I think it's safer to use an `empty` check here as it catches both cases where the list is empty and potentially if it returns a false. Only because there's no specific handling for the `false` return (e.g. logging this info) so it doesn't just blow up on the front end :)